app/services/article_api_index_service.rb

```diff
@@ -64,6 +66,10 @@ def tag_articles
     articles.page(page).per(30)
   end
 
+  def top_articles
+    Article.published.order("positive_reactions_count DESC").where("published_at > ?", top.to_i.days.ago)
```

Comment:
You might want to provide a limit for top, using pagination like in `username_articles` because, if I'm not mistaken, I could actually get all published articles in one request by going back thousands of days.

Reply:
Oh yeah, thanks for catching that!